POEM_052.md

```diff
@@ -267,23 +368,27 @@ The matrix vector product method method signature will expect three additional a
 def jac_vec_some_func(x, y, z, d_inputs, d_outputs, mode):
     ...  
 
+@om.func.compute_jcavec_product(jac_vec_some_func)
 def some_func(x:{'units':'m'}=np.zeros(4),
               y:{'units':'m'}=np.ones(4),
               z:{'units':None}=3) 
-              -> [('foo',{'units':1/m, 'shape':4}), ('bar',{'units':'m', 'shape':4}), 
+               -> dict(foo={'units':1/m, 'shape':4},
+                      bar={'units':'m', 'shape':4}):
                   ('compute_jacvec_product', jac_vec_some_func), 
                  ] : 
 
     foo = np.log(z)/(3*x+2*y)
     bar = 2*x+y
     return foo, bar
 
-comp = om.ExplicitFuncComp(some_func,)    
+some_func_comp = om.ExplicitFuncComp(some_func,)    
 ```
 
 
 ## Implicit API Description
 
+Note: I want to look at this more, but it seems to me that since ImplicitComponents require so many additional methods, I would rather use the class syntax; I think supporting all of the declarations that would be necessary to construct typical ImplicitComponents is a lot of work without much payoff. If we can use some of the syntax sugar above to reduce boilerplate for the `initialize` and `setup` method definitions, it may be worth looking into that.
```

Comment:
its one extra method, that is itself actually optional. 

Explicit components need: compute
Implicit components need: apply_nonlinear, solve_nonlinear (optional). 

I don't understand your objection here. A user who prefers the functional API will surely prefer it for both implicit and explicit functions. Also, one of the key motivations is to make things more AD-able. Both implicit and explicit functions should be allowed.

Reply:
This is a stale note, sorry for the confusion. I agree we should be able to any pure python functions to fulfill the component interfaces as needed, with a clean, DRY interface for wrapping arguments/returns. Will resolve along with jacobian.